python/paddle/fluid/clip.py

```diff
@@ -264,15 +264,15 @@ def _process_context(self, context, param, grad):
             context[self.group_name] = []
             context[self.group_name + "_clip_value"] = self.clip_norm
             context[self.group_name + "_clip"] = layers.fill_constant(
-                shape=[1], dtype="float32", value=self.clip_norm)
+                shape=[1], dtype=grad.dtype, value=self.clip_norm)
         else:
             if not self.clip_norm == context[self.group_name + "_clip_value"]:
                 raise ValueError(
                     "All parameters' 'clip_norm' of a same group should be the same"
                 )
 
-        square = grad * grad
-        local_norm_var = layers.cast(layers.reduce_sum(input=square), 'float64')
+        square = layers.pow(grad, 2.0)
+        local_norm_var = layers.reduce_sum(input=square)
```

Comment:
float64 is no longer needed?

Reply:
I am not sure, the original code of line 275 is written by @ktlichkid, can you explain why use `float64` here?

Reply:
The computation here sums the square of gradients up globally, thus it may easily cause overflow. Float64 here is to prevent such overflows, and after the square root computation this value is again cast back to float32.